src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -909,54 +909,61 @@ private static ReadOnlySpan<char> EatAllWhitespace(ReadOnlySpan<char> str)
         // Returns an unsigned byte array containing the GUID.
         public byte[] ToByteArray()
         {
-            var g = new byte[16];
+            byte[] array = new byte[16];
+
             if (BitConverter.IsLittleEndian)
             {
-                MemoryMarshal.TryWrite(g, in this);
+                Unsafe.WriteUnaligned(ref MemoryMarshal.GetArrayDataReference(array), this);
```

Comment:
does it make it any better? seems to be more verbose + bonus points for extra "Unsafe" word.

Reply:
@EgorBo updated description with a diff showing bounds check elimination

Reply:
I think it's better to extract the problematic part and file as an issue against JIT rather than workarounding it

Reply:
`MemoryMarshal.TryWrite` into a newly created `new byte[16]` shouldn't produce any checks, if it does - jit should be improved.

Reply:
The duplicate bounds check goes away with the addition of a `uint` cast here:

https://github.com/dotnet/runtime/blob/0378936909464c84cf207ffd1a21efa474fc34c0/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L968